### PR TITLE
Remove usage of Confluence wiki

### PR DESCRIPTION
**Pull request by scoheb**
- rely on plugins.jenkins.io

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -54,15 +54,15 @@
...
-      </plugin>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>exec-maven-plugin</artifactId>
+        <version>1.2</version>
+        <configuration>
+          <executable>java</executable>
+          <mainClass>${main.class}</mainClass>
+        </configuration>
+      </plugin>
```
Comment (ikedam) [inline pom.xml]: (Just note for me) Changed CRLF -> LF.

**Comment by ikedam** [comment]
This is a porting of following commits:
* jenkins-infra/backend-update-center2@39403d5cf9442b16b30492812fc6309979aaf1a4
    * Omitting `-urlmap` (`pluginDocumentationUrl`) feature.
* jenkins-infra/backend-update-center2@97e070138180cba9a728617ca468be5fe0f98813

File: src/main/java/org/jvnet/hudson/update_center/Main.java [hunk 2]
```diff
@@ -389,7 +376,8 @@ protected JSONArray buildReleaseHistory(MavenRepository repository) throws Excep
                     o.put("title", title);
                     o.put("gav", h.artifact.groupId+':'+h.artifact.artifactId+':'+h.artifact.version);
                     o.put("timestamp", h.getTimestamp());
-                    o.put("wiki", plugin.getWiki());
+                    o.put("wiki", plugin.getPluginUrl());
+                    o.put("url", "https://plugins.jenkins.io/" + h.artifact.artifactId);
```
Comment (ikedam) [inline src/main/java/org/jvnet/hudson/update_center/Main.java]: 1539d22e615f5fcca97e17e6ba07fcb98612f876@jenkins-infra/backend-update-center2
I'm not sure this is actually needed (Rather I'm not sure release-history.json is actually used).
This looks a change unrelated to the purpose of this request, but it's OK.

File: src/main/java/org/jvnet/hudson/update_center/Plugin.java [hunk 3]
```diff
@@ -69,43 +67,23 @@
...
-     * Null if we couldn't find it.
-     */
-    public final RemotePage page;
-
-    /**
-     * Confluence labels for the plugin wiki page.
-     * Null if wiki page wasn't found.
-     */
-    private String[] labels;
-    private boolean labelsRead = false;
```
Comment (ikedam) [inline src/main/java/org/jvnet/hudson/update_center/Plugin.java]: `labels` and `labelsRead` are still left in the upstream, but actually they are no longer used.
That looks rather a bug of the upstream. (actually not a bug as that doesn't cause problems, but I don't know an appropriate word for this situation)
Please consider to contribute also to the upstream if you can.

File: src/main/java/org/jvnet/hudson/update_center/Plugin.java [hunk 4]
```diff
@@ -69,43 +67,23 @@
...
-    private final Document pom;
+    private Document pom;
 
-    public Plugin(String artifactId, HPI latest, HPI previous, ConfluencePluginList cpl) throws IOException {
+    public Plugin(String artifactId, HPI latest, HPI previous) throws IOException {
         this.artifactId = artifactId;
         this.latest = latest;
         this.previous = previous;
         this.xmlReader = createXmlReader();
         this.pom = readPOM();
```
Comment (ikedam) [inline src/main/java/org/jvnet/hudson/update_center/Plugin.java]: `pom` should be initialized lazily as you are porting 97e070138180cba9a728617ca468be5fe0f98813@jenkins-infra/backend-update-center2

Comment (scoheb) [inline src/main/java/org/jvnet/hudson/update_center/Plugin.java]: Not quite sure I understand what change you'd like me to make.

File: src/main/java/org/jvnet/hudson/update_center/Main.java (hunk 2 above)
Comment (scoheb) [inline src/main/java/org/jvnet/hudson/update_center/Main.java]: I'd rather not remove such a feature in this PR.

File: src/main/java/org/jvnet/hudson/update_center/Plugin.java (hunk 3 above)
Comment (scoheb) [inline src/main/java/org/jvnet/hudson/update_center/Plugin.java]: Created https://github.com/jenkins-infra/backend-update-center2/pull/133 #

Comment (ikedam) [inline src/main/java/org/jvnet/hudson/update_center/Plugin.java]: Thanks a lot!

File: src/main/java/org/jvnet/hudson/update_center/Plugin.java (hunk 4 above)
Comment (ikedam) [inline src/main/java/org/jvnet/hudson/update_center/Plugin.java]: `this.pom = readPOM()` should be removed and left being `null`.

**Comment by scoheb** [comment]
Changes made. Let me know what you think. Tested here locally and update_center.json is populated correctly.

**Comment by scoheb** [comment]
@ikedam 
I have fixed a couple of issues. Hopefully you can take a look?

**Comment by ikedam** [comment]
Would you tell me an actual case that 809b785d1822554e5a33a9196b1e4bdc3a5e42da works?
We might better port it to the upstream if it occurs with official plugins.

**Comment by scoheb** [comment]
gitlab-hook is one of them
See http://mirrors.jenkins-ci.org/plugins/gitlab-hook/1.4.2/gitlab-hook.hpi

**Comment by ikedam** [comment]
The additional change is essentially same to #23 (rather, the new change is better than that as it also refers MANIFEST.MF).
It looks good to me.

File: src/main/java/org/jvnet/hudson/update_center/Plugin.java [hunk 5]
```diff
@@ -331,29 +269,18 @@ public JSONObject toJSON() throws IOException {
...
 
-        if (!json.has("excerpt")) {
-            // fall back to <description>, which is plain text but still better than nothing.
-            if(pom != null)
-            {
-                String description = plainText2html(selectSingleValue(pom, "/project/description"));
-                if (description!=null)
-                    json.put("excerpt",description);
-            }
+        json.put("wiki", "https://plugins.jenkins.io/" + artifactId);
```
Comment (ikedam) [inline src/main/java/org/jvnet/hudson/update_center/Plugin.java]: This results URLs even of private plugins get https://plugins.jenkins.io/... .
I added fab26b4e7afb7520690fd3024329989bade0cbb6 to have "-nowiki" switch disabled it and use the URL in pom.xml instead.

**Pull request merged**